posts/2021-02-15-eclipse-transformer.adoc

```diff
@@ -0,0 +1,506 @@
+
+To solve this problem, IBM development created the Eclipse Transformer, an open source project which provides a java utility that updates package references in a variety of artifacts.
+
+Most immediately, IBM uses the Eclipse Transformer to transform bundle implementations archives, quickly creating Jakarta EE 9 implementation bundle archives from existing Java EE archives.
+
+But, that is a starting point.  Several additional types of resources must be transformed:
+
+* Test applications;
+* The actual tests;
+* Server configuration files used by the tests;
+
+This article describes how developers may, in a similar fashion, use the Eclipse Transformer update their application and test artifacts to enable these to run using Jakarta EE 9.
+
+== Transformation Strategy
+
+The stratgy used by Open-Liberty builds is to annotate test classes and methods to indicate which classes and methods are to be run as Jakarta EE 9 tests.  Then, when running tests, if any tests are marked to be run as Jakarta EE 9 tests, the test are re-run, first performing a JavaEE to Jakarta EE 9 transformation on the test artifacts, then actually running the transformed tests.
+
+Two specific updates are required by test artifacs.  First, existing package references are updated, changing JavaEE package references to Jakarta EE 9 references, for those packages which are present in Jakarta EE 9 specifications.  Package references are in test applications, test code, and in server configurations.  Second, server configurations are updated to replace JavaEE features with the corresponding Jakarta EE 9 features.  The Eclipse Transformer may be used to perform the first of these updates.  Server configuration features must be updated manually.
+
+== Invoking the Transformer
+
+The Eclipse-Transformer can be used directly from the command line, or can be invoked as a build task, or directly from java code.  This article describes using the transformer from the command line.  See the transformer project pages for additional usage guidelines.  
+
+=== The Eclipse Transformer Project
+
+The Eclipse Transformer is an Eclipse hosted open source project:
+
+[source]
+--
+https://projects.eclipse.org/projects/technology.transformer
+--
+
+=== Downloading the Transformer 
+
+A latest snapshot of the Eclipse Transformer may be obtained, for example, as:
+
+https://oss.sonatype.org/content/repositories/snapshots/org/eclipse/transformer/org.eclipse.transformer.cli/0.3.0-SNAPSHOT/org.eclipse.transformer.cli-0.3.0-20201027.133230-4-distribution.jar
+
+=== Building the transformer:
```

Comment:
The other headings do not end with a colon (`:`). This one shouldn't so that there is consistency.